contrib/platform/src/com/sun/jna/platform/mac/IOKit.java

```diff
@@ -0,0 +1,418 @@
+     */
+    int IORegistryEntryCreateCFProperties(long entry, PointerByReference properties, CFAllocatorRef allocator,
+            int options);
+
+    /**
+     * Create a CF representation of a registry entry's property.
+     *
+     * @param entry
+     *            The registry entry at which to start the search.
+     * @param plane
+     *            The name of an existing registry plane. Plane names are defined in
+     *            {@code IOKitKeys.h}, for example, {@code kIOServicePlane}.
+     * @param key
+     *            A {@code CFString} specifying the property name.
+     * @param allocator
+     *            The CF allocator to use when creating the CF container.
+     * @param options
+     *            {@link #kIORegistryIterateRecursively} may be set to recurse
+     *            automatically into the registry hierarchy. Without this option,
+     *            this method degenerates into the standard
+     *            {@link #IORegistryEntryCreateCFProperty} call.
+     *            {@link #kIORegistryIterateParents} may be set to iterate the
+     *            parents of the entry, in place of the children.
+     * @return A CF container is created and returned the caller on success. The
+     *         caller should release with CFRelease.
+     */
+    CFTypeRef IORegistryEntrySearchCFProperty(long entry, String plane, CFStringRef key, CFAllocatorRef allocator,
+            int options);
+
+    /**
+     * Returns an ID for the registry entry that is global to all tasks.
+     *
+     * @param entry
+     *            The registry entry handle whose ID to look up.
+     * @param id
+     *            The resulting ID.
+     * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+     */
+    int IORegistryEntryGetRegistryEntryID(long entry, LongByReference id);
+
+    /**
+     * Returns a name assigned to a registry entry.
+     *
+     * @param entry
+     *            The registry entry handle whose name to look up.
+     * @param name
+     *            The caller's buffer to receive the name.
+     * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+     */
+    int IORegistryEntryGetName(long entry, Pointer name);
+
+    /**
+     * Returns an iterator over a registry entry’s child entries in a plane.
+     *
+     * @param entry
+     *            The registry entry whose children to iterate over.
+     * @param plane
+     *            The name of an existing registry plane. Plane names are defined in
+     *            {@code IOKitKeys.h}, for example, {@code kIOServicePlane}.
+     * @param iter
+     *            The created iterator over the children of the entry, on success.
+     *            The iterator must be released when the iteration is finished.
+     * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+     */
+    int IORegistryEntryGetChildIterator(long entry, String plane, LongByReference iter);
```

Comment:
io_name_t is a fixed size array, so will this work with string binding?

Reply:
Well, [it's been working for 3+ years.](https://github.com/oshi/oshi/blob/5024f56c1909b7977801b658317ddb47bb7aaec1/oshi-core/src/main/java/oshi/hardware/platform/mac/MacUsbDevice.java#L145)... doesn't mean it's correct, though.  

But the documentation specifies that all these keys are defined in [`IOKitKeys.h`](https://opensource.apple.com/source/xnu/xnu-201/iokit/IOKit/IOKitKeys.h.auto.html) where the defines look like Strings:
```
// IOService class name
#define kIOServiceClass			"IOService"
```
So I'm thinking this should work.